src/sourmash/index/__init__.py

```diff
@@ -701,31 +706,42 @@ def select(self, **kwargs):
 
 
 class CounterGather:
-    """
-    Track and summarize matches for efficient 'gather' protocol.  This
-    could be used downstream of prefetch (for example).
+    """This is an ancillary class that is used to implement "fast
+    gather", post-prefetch. It tracks and summarize matches for
+    efficient min-set-cov/'gather'.
+
+    The class constructor takes a query MinHash that must be scaled, and
+    then takes signatures that have overlaps with the query (via 'add').
+
+    After all overlapping signatures have been loaded, the 'peek'
+    method is then used at each stage of the 'gather' procedure to
+    find the best match, and the 'consume' method is used to remove
+    a match from this counter.
 
-    The public interface is `peek(...)` and `consume(...)` only.
+    This particular implementation maintains a collections.Counter that
+    is used to quickly find the best match when 'peek' is called, but
+    other implementations are possible ;).
     """
     def __init__(self, query_mh):
+        "Constructor - takes a query FracMinHash."
```

Comment:
Is it limited to accepting FracMinHash? If so, would you please point me to where I can find why it's limited to FracMinHashes?

Also, if it is limited to FracMinHashes, do you think we should add an if condition?

Reply:
Yep - it's limited to estimators that support containment operations, and num MinHashes do not.

The if statement a few lines below does the correct check:
```python
        if not query_mh.scaled:
            raise ValueError('gather requires scaled signatures')
```
which is tested by `test_counter_gather_num_query`.

In addition, (spoiler alert!) https://github.com/sourmash-bio/sourmash/pull/2116 refactors the class to take a `SourmashSignature` instead of a `MinHash`, which (in some possible future) opens it up to other kinds of estimators.